src/auto_controller/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 3.5.1)
+project(auto_controller)
+
+set(SOURCE_FILES
+  src/auto_controller_core.cpp
+  src/recording_controller.cpp
+  )
+
+add_library(${PROJECT_NAME} STATIC ${SOURCE_FILES})
+
+include_directories(${MODULE_INCLUDE_DIRS} include)
+target_include_directories(${PROJECT_NAME} PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/ext/span/include>
+  )
+target_link_libraries(auto_controller jsoncpp)
```

Comment:
Why is this a separate project? I don't see a need for logical module separation here, you can move all your code into `controller`